### ISSUE TITLE
Complex values raise uncaught exception

### ISSUE THREAD

**Issue by ghiander** [AstraZeneca/jazzy#382]
**Describe the bug**
Organometals trigger an uncaught exception:

**To Reproduce**
```
from jazzy.api import molecular_vector_from_smiles
s = "C12C3C4C5C1[Fe]23456789C%10C6C7C8C9%10"
molecular_vector_from_smiles(s, minimisation_method=None)

> (jazzy39) (base) [~/jazzy_test: (master)]:$ python jazzy_test.py 
[10:29:06] UFFTYPER: Unrecognized hybridization for atom: 5
[10:29:06] UFFTYPER: Unrecognized atom type: Fe+2 (5)
/.../src/jazzy/core.py:513: ComplexWarning: Casting complex values to real discards the imaginary part
  return float(gd * don + ga * acc)
Traceback (most recent call last):
  File "/home/ghiander/.../jazzy_test.py", line 17, in <module>
    molecular_vector_from_smiles(s, minimisation_method=None)
  File "/.../src/jazzy/api.py", line 90, in molecular_vector_from_smiles
    dg = {k: round(dg[k], ROUNDING_DIGITS) for k in dg}
  File "/.../src/jazzy/api.py", line 90, in <dictcomp>
    dg = {k: round(dg[k], ROUNDING_DIGITS) for k in dg}
TypeError: type complex doesn't define __round__ method
```

**Expected behavior**
Either reject the compound or discard the imaginary part, parse as float, and let the rounding happen.
More investigation is needed to understand how a complex number is generated when running this example.
Creation of test for this example if mandatory.

**Comment by ghiander** [comment]
`acc` results being a complex number due to the presence of the Fe atom which has a negative number of lone pairs (-1). `acc` is the result from `nlps**expa` == -1 ** 0.34 == -1 ** 17/50 which results in a complex number.

I created an exception `NegativeLonePairsError()` that should be raised in `core.py` when atoms with negative lone pairs come across. I am a bit less sure on what to do from an API stand point. I think the API should instead catch any specific errors and convert them into JazzyError to simplify integration and avoid breaking existing ones when people upgrade to a new version of Jazzy.

@f3rmion - thoughts?

**Comment by f3rmion** [comment]
> I created an exception NegativeLonePairsError() that should be raised in core.py when atoms with negative lone pairs come across. 

Agreed, we need to catch those errors explicitly. 

> I am a bit less sure on what to do from an API stand point. I think the API should instead catch any specific errors and convert them into JazzyError to simplify integration and avoid breaking existing ones when people upgrade to a new version of Jazzy.

Let's wrap the `NegativeLonePairsError` and raise a `JazzyError` in the API. This way the root cause is still visible.

**Comment by ghiander** [comment]
@f3rmion https://github.com/AstraZeneca/jazzy/pull/401

I have created the specific error, added a decorator for the API functions, and included tests. 100% coverage. Please review

**Comment by f3rmion** [comment]
Fixed by #401
